src/resource/ResourceManager.cpp

```diff
@@ -22,7 +22,7 @@ ResourceManager::ResourceManager(const std::string& mainPath, const std::string&
     size_t threadCount = 1;
 #else
     // the extra `- 1` is because we reserve an extra thread for spdlog
-    size_t threadCount = std::max(1U, (std::thread::hardware_concurrency() - reservedThreadCount - 1));
+    size_t threadCount = std::max(1, (int)(std::thread::hardware_concurrency() - reservedThreadCount - 1));
```

Comment:
Change reservedThreadCount to an int32_t. Also should be using int32_t rather than int.

Reply:
What would changing reservedThreadCount to int32 accomplish?

Reply:
You want to work in signed because the underflow in unsigned doesn't allow you to test correctly. It accomplishes working in the correct data range.

Also, as mentioned, we use int32_t rather than int.

Reply:
I didn't mean that to seem like I was asking why I should bother, it just didn't seem like it would change anything functionally.